easybuild/easyblocks/g/gromacs.py

```diff
@@ -324,34 +375,60 @@ def configure_step(self):
                     if not regex.search(out):
                         raise EasyBuildError("Pattern '%s' not found in GROMACS configuration output.", pattern)
 
+    def build_step(self):
+        """
+        Custom build step for GROMACS; Skip if CUDA is enabled and the current
+        iteration is for double precision
+        """
+
+        cuda = get_software_root('CUDA')
+        if cuda:
+            if re.search('-DGMX_DOUBLE=(ON|YES|TRUE|Y|[1-9])', self.cfg.get('configopts'), re.I):
+                self.log.info("skipping build step")
+                return
+
+        super(EB_GROMACS, self).build_step()
+
     def test_step(self):
         """Run the basic tests (but not necessarily the full regression tests) using make check"""
         # allow to escape testing by setting runtest to False
-        if not self.cfg['runtest'] and not isinstance(self.cfg['runtest'], bool):
+        if not self.cfg.get('runtest') and not isinstance(self.cfg.get('runtest'), bool):
 
             # make very sure OMP_NUM_THREADS is set to 1, to avoid hanging GROMACS regression test
             env.setvar('OMP_NUM_THREADS', '1')
 
             self.cfg['runtest'] = 'check'
-            if self.cfg['parallel']:
+            if self.cfg.get('parallel'):
                 # run 'make check' in parallel since it involves more compilation
-                self.cfg.update('runtest', "-j %s" % self.cfg['parallel'])
+                self.cfg.update('runtest', "-j %s" % self.cfg.get('parallel'))
             super(EB_GROMACS, self).test_step()
 
+            # Set runtest to None so that the gmxapi extension doesn't try to
+            # run "check" as a command
+            self.cfg['runtest'] = None
+
     def install_step(self):
         """
         Custom install step for GROMACS; figure out where libraries were installed to.
-        Also, install the MPI version of the executable in a separate step.
         """
+        # Skipping the install step if CUDA is enabled and the current iteration
+        # is for double precision
+        cuda = get_software_root('CUDA')
```

Comment:
Ok after looking at this again: You have the same, non-trivial check at 3 places (configure, build, install)

I can't remember if you can simply omit the double-precision case for CUDA when generating the variant steps, that would be best. Second best: Factor the check into a function like `is_double_precision_cuda_build`   
That would also make the comment unnecessary.

BTW: I think the `re.I` is not required or even wrong. The part `-DGMX_DOUBLE=` has to be upper-case and to be fully correct for the remaining: Everything that is not false is true in CMake, so that check is incomplete anyway. But what you mainly want to check is for `-DGMX_DOUBLE=ON` which you add in https://github.com/easybuilders/easybuild-easyblocks/pull/1991/files#diff-a02b2451ce7221a206c1f64ba6758d33R594. So  `if '-DGMX_DOUBLE=ON' in  self.cfg.get('configopts', '')` should be enough. (or `self.cfg['configopts']` if we can assume it's always there)

If an EC author adds something different: Not your fault ;)

Reply:
Yeah, should probably filter out enable-double|DGMX_DOUBLE from configopts coming from the EC but...

Reply:
Speaking of that: You should probably use `-DGMX_DOUBLE=OFF` for the single precision case or an EC passing `-DGMX_DOUBLE=...` will still make it compile as double. But besides that I'd say: Assume nothing is coming from the EC or it is at least well-behaved (using ON/OFF)